torch/_refs/__init__.py

```diff
@@ -830,6 +839,34 @@ def sub(
     aten_op=None,  # CompositeImplicitAutograd
 )
 
+
+def _xlogy(a: Union[TensorLikeType, NumberType], b: Union[TensorLikeType, NumberType]):
+    assert isinstance(a, TensorLike) or isinstance(b, TensorLike)
+
+    # torch.xlogy supports scalar inputs but torch.log does not.
+    # TODO Add support for scalar inputs to refs.log (and other elementwise unary ops)
```

Comment:
So you need to only wrap `b` if the problem is `log`? And even in that case, you shouldn't be putting it on device, `mul` should be able to handle cpu 0-dim input?

Reply:
`mul` handles a cpu scalar. However, it is the `isnan(b)` condition in the `where` that causes the issue.